container/apptabs.go

```diff
@@ -277,18 +277,22 @@ type appTabsRenderer struct {
 
 func (r *appTabsRenderer) Layout(size fyne.Size) {
 	// Try render as many tabs as will fit, others will appear in the overflow
-	for i := len(r.appTabs.Items); i > 0; i-- {
-		r.updateTabs(i)
-		barMin := r.bar.MinSize()
-		if r.appTabs.location == TabLocationLeading || r.appTabs.location == TabLocationTrailing {
-			if barMin.Height <= size.Height {
-				// Tab bar is short enough to fit
-				break
-			}
-		} else {
-			if barMin.Width <= size.Width {
-				// Tab bar is thin enough to fit
-				break
+	if len(r.appTabs.Items) == 0 {
+		r.updateTabs(0)
+	} else {
+		for i := len(r.appTabs.Items); i > 0; i-- {
+			r.updateTabs(i)
+			barMin := r.bar.MinSize()
+			if r.appTabs.location == TabLocationLeading || r.appTabs.location == TabLocationTrailing {
+				if barMin.Height <= size.Height {
+					// Tab bar is short enough to fit
+					break
+				}
+			} else {
+				if barMin.Width <= size.Width {
+					// Tab bar is thin enough to fit
+					break
+				}
 			}
 		}
 	}
```

Comment:
We prefer to have the main path not indented and instead keep any special cases in their own if-statements with a return statement instead. Something like this:
```go
	if len(r.appTabs.Items) == 0 {
		r.updateTabs(0)
		return
	}
	
        for i := len(r.appTabs.Items); i > 0; i-- {
	        r.updateTabs(i)
	        barMin := r.bar.MinSize()
	        if r.appTabs.location == TabLocationLeading || r.appTabs.location == TabLocationTrailing {
		        if barMin.Height <= size.Height {
			        // Tab bar is short enough to fit
			        break
		        }
	        } else {
		        if barMin.Width <= size.Width {
			        // Tab bar is thin enough to fit
			        break
		        }
	        }
        }
```

Reply:
@Jacalz Thanks for reviewing. It looks clear in this style. But we cannot return in this position, we need excute the rest code after `updateTabs`, so i wrapped for-loop in if-else statement even it looks not pretty.
https://github.com/fyne-io/fyne/blob/79968626fe10c3cd49b978bc2f0d635bc8e18d31/container/apptabs.go#L278-L305

Reply:
Ah, I see now. Sorry about that. It seems like GitHub didn't display that part at all in the diff and didn't make it very clear that the function was longer than that.

Reply:
it doesn't matter :)

Reply:
Cool. I'll review this later today and approve if everything's working fine :)